ptf/saimpls.py

```diff
@@ -0,0 +1,3818 @@
+            sai_thrift_remove_inseg_entry(self.client, inseg_entry_4)
+            sai_thrift_remove_debug_counter(self.client, debug_cnt)
+
+
+@group("draft")
+class MplsIpv6Test(SaiHelper):
+    ''' Basic IPv6 MPLS test class '''
+    def setUp(self):
+        super(MplsIpv6Test, self).setUp()
+
+        ttl_value = 63
+        exp_value = 5
+        ipv6_1 = '100::1'
+        ipv6_2 = '100::2'
+        ipv6_3 = '100::3'
+        ipv6_4 = '220::1'
+        ipv6_5 = '330::1'
+        ipv6_6 = '550::1'
+        ipv6_7 = '700::1'
+        ipv6_8 = '710::1'
+        ipv6_9 = '720::1'
+        ipv6_10 = '800::1'
+
+        ipv6_subnet_1 = '110::1/128'
+        ipv6_subnet_2 = '120::1/128'
+        ipv6_subnet_3 = '130::1/128'
+        ipv6_subnet_4 = '200::1/128'
+        dmac = '00:11:22:33:44:55'
+
+        # Ingress LER config
+        self.ingress_rif = self.port10_rif
+        self.egress_rif_1 = self.port11_rif
+
+        label_list = [1000]
+        label_list_t = sai_thrift_u32_list_t(
+            count=len(label_list),
+            uint32list=label_list)
+
+        self.nhop_label_1 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress(ipv6_1),
+            router_interface_id=self.egress_rif_1,
+            type=SAI_NEXT_HOP_TYPE_MPLS,
+            labelstack=label_list_t,
+            outseg_ttl_mode=SAI_OUTSEG_TTL_MODE_PIPE,
+            outseg_ttl_value=ttl_value,
+            outseg_exp_value=exp_value,
+            outseg_type=SAI_OUTSEG_TYPE_PUSH)
+
+        label_list.append(2000)
+        label_list_t = sai_thrift_u32_list_t(
+            count=len(label_list),
+            uint32list=label_list)
+
+        self.nhop_label_2 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress(ipv6_2),
+            router_interface_id=self.egress_rif_1,
+            type=SAI_NEXT_HOP_TYPE_MPLS,
+            labelstack=label_list_t,
+            outseg_ttl_mode=SAI_OUTSEG_TTL_MODE_PIPE,
+            outseg_ttl_value=ttl_value,
+            outseg_exp_value=exp_value,
+            outseg_type=SAI_OUTSEG_TYPE_PUSH)
+
+        label_list.append(3000)
+        label_list_t = sai_thrift_u32_list_t(
+            count=len(label_list),
+            uint32list=label_list)
+
+        self.nhop_label_3 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress(ipv6_3),
+            router_interface_id=self.egress_rif_1,
+            type=SAI_NEXT_HOP_TYPE_MPLS,
+            labelstack=label_list_t,
+            outseg_ttl_mode=SAI_OUTSEG_TTL_MODE_PIPE,
+            outseg_ttl_value=ttl_value,
+            outseg_exp_value=exp_value,
+            outseg_type=SAI_OUTSEG_TYPE_PUSH)
+
+        # note: nbr with the same dst_ip as mpls nhop has to be created
+        # after mpls nhop is created,
+        # if not it will create second nhop of type IP with this dst_ip
+        # and neither of them will be functioning correctly - packets
+        # will be gleaned to CPU
```

Comment:
minor: Please use a common convention for abbreviations in comments and use capital letters for them. Also, I would recommend to use full names for nbr and nhop